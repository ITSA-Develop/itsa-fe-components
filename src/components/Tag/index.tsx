import { Tag as AntTag, TagProps } from 'antd';

export const Tag = ({ ...rest }: TagProps) => {
	return <AntTag {...rest} />;
}