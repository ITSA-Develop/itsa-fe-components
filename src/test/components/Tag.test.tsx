import { render, screen } from '@testing-library/react';
import { Tag } from '../../components/Tag/Tag';

describe('Tag component', () => {
	it('renders with children and additional props', () => {
		const { container } = render(<Tag color="blue">Test Tag</Tag>);
		const tagElement = screen.getByText('Test Tag');
		expect(tagElement).toBeInTheDocument();
		expect(tagElement).toHaveClass('ant-tag-blue');
		expect(container).toMatchSnapshot();
	});

	it('passes extra props to AntTag', () => {
		const { container } = render(
			<Tag id="custom-id" data-testid="tag-testid">
				Content
			</Tag>,
		);
		const tag = screen.getByTestId('tag-testid');
		expect(tag).toBeInTheDocument();
		expect(tag.id).toBe('custom-id');
		expect(container).toMatchSnapshot();
	});
});
