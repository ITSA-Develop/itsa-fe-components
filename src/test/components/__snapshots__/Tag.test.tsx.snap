// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag component > passes extra props to AntTag 1`] = `
<div>
  <span
    class="ant-tag css-dev-only-do-not-override-l9pxc0"
    data-testid="tag-testid"
    id="custom-id"
  >
    Content
  </span>
</div>
`;

exports[`Tag component > renders with children and additional props 1`] = `
<div>
  <span
    class="ant-tag ant-tag-blue css-dev-only-do-not-override-l9pxc0"
  >
    Test Tag
  </span>
</div>
`;
